src/include/error.h

```diff
@@ -97,15 +97,24 @@
     } while (0)
 #define WT_RET_BUSY_OK(a) WT_RET_ERROR_OK(a, EBUSY)
 #define WT_RET_NOTFOUND_OK(a) WT_RET_ERROR_OK(a, WT_NOTFOUND)
-/* Set "ret" if not already set. */
-#define WT_TRET(a)                                                                           \
-    do {                                                                                     \
-        int __ret;                                                                           \
-        if ((__ret = (a)) != 0 &&                                                            \
-          (__ret == WT_PANIC || ret == 0 || ret == WT_DUPLICATE_KEY || ret == WT_NOTFOUND || \
-            ret == WT_RESTART))                                                              \
-            ret = __ret;                                                                     \
+/*
+ * Set a return code if one is not already set. Upgrade the existing return code if the new one is
+ * more urgent. Return codes should be saved in a variable named "ret" wherever possible. Prefer
+ * using WT_TRET over this macro.
+ */
+#define WT_TRET_FUNC(ret, a)                                                                       \
```

Comment:
I don't particularly like the semantics of this - the urgency of an error code can change with context. It's also not clear exactly what this macro will do when using it, even after reading the comment. Looking at the callsites - I suspect this would no longer be necessary if we return errors from the new walk function?

Reply:
Resolving as the error code return method negates this issue.